wcfsetup/install/files/lib/system/WCF.class.php

```diff
@@ -532,11 +535,29 @@ protected function initTPL()
     protected function initStyle()
     {
         if (isset($_REQUEST['styleID'])) {
-            self::getSession()->setStyleID(\intval($_REQUEST['styleID']));
+            $styleID = \intval($_REQUEST['styleID']);
+
+            try {
+                $action = new StyleAction([$styleID], 'changeStyle');
+                $action->validateAction();
+                $action->executeAction();
+            } catch (PermissionDeniedException | UserInputException $e) {
+                if (self::getSession()->getUser()->userID) {
+                    $styleID = self::getSession()->getUser()->styleID ?: 0;
+                } else {
+                    $styleID = self::getSession()->getVar('styleID') ?: 0;
+                }
```

Comment:
It would be simpler initializing `$styleID = 0`, then doing `$styleID = 0` here in this catch and replacing the `else` by `if (!$styleID)`.